### PR TITLE
(SIMP-4384) Add an oscap fact

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Added an `oscap` fact to collect the following information:
  - OpenSCAP Version
  - OpenSCAP Supported Specifications
  - SCAP Profiles from /usr/share/xml/scap/*/content/*-ds.xml

SIMP-4773 #close

### CODE REVIEW

File: lib/facter/oscap.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    version_info = Facter::Core::Execution.execute("#{oscap} version")
+
+    # Get the version of oscap on the system
+    if version_info =~ %r{^\s*OpenSCAP command line tool.*(\d+\.\d+\.\d+)$}
+      retval['version'] = $1.strip
+    end
+
+    # Now get the supported versions of the various specs
+    if version_info =~ %r{=+ Supported specifications =+(.*?)(^\s*$|=+)}m
+      retval['supported_specifications'] = Hash[$1.lines.map do |l|
```
Comment (jeefberkey) [inline lib/facter/oscap.rb]: This `Hash[` syntax is hard to follow

File: spec/unit/facter/oscap_spec.rb [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+      Facter::Core::Execution.stubs(:execute).with('/bin/oscap info --profiles /usr/share/xml/scap/ssg/content/ssg-centos7-ds.xml').returns(<<-EOM
+xccdf_org.ssgproject.content_profile_standard:Standard System Security Profile
+        EOM
+     )
+
+      value = Facter.fact(:oscap).value
+      expect(value['profiles']).to be_a(Hash)
+
+      expect(value['profiles']).to be_a(Hash)
```
Comment (jeefberkey) [inline spec/unit/facter/oscap_spec.rb]: Dup test

**Comment by trevor-vaughan** [comment]
@lnemsick-simp That's part of SIMP-4774. I wanted to make sure nobody hated this layout prior to doing the rest of the ticket.

**Comment by trevor-vaughan** [comment]
Don't push this until the next patch comes in. Ran into issues on old versions of EL6

**Comment by trevor-vaughan** [comment]
:heavy_check_mark: https://gitlab.com/simp/pupmod-simp-openscap/pipelines/24606089
